### DISCUSSION TITLE
supporting both hostbased and routebased strategy

### DISCUSSION THREAD

**Discussion by vigouredelaruse** [Finbuckle/Finbuckle.MultiTenant#560] [Q&A] [answered]
i am implementing a multitenant ~~cms~~ site development framework with relative ease thanks to this library so kudos

now suppose i want to simultaneously support typical wordpress site urls like
- sometenant.somehostingprovider.com/tenanturlspace
- www.somehostingprovider.com/sometenant/tenanturlspace

question: how do i configure routes for this combined scenario?

right now i have hardcoded {__ tenant __} in controller routes but i'm confused about what happens when the matched strategy is not pathbased or route. consider the following code
```
                options.MapControllerRoute(
                  name: "default",
                  pattern: "{controller=Home}/{action=Index}/{id?}");

                options.MapControllerRoute(
                  name: "multitenantdefault",
                  pattern: "{__tenant__}/{controller}/{action}");

somecontroller
               [Route("{__tenant__}/api/[controller]")]

services.AddMultiTenant<TenantInfo>()
        .WithRouteStrategy() // or should this come after .withHost() for this use case?
        .WithHostStrategy("*.__tenant__.?.?")
```

now consider an ambiguous request, where a url path matches 2 different tenants
https://matchedtenant.domain.com/innocentPathThatMatchesATenant

please help me demystify the following: given the above routes and multitenant registrations,  
- to support both route and host strategies is it necessary that one needs to apply multiple [Route] in controllers?

regards

**Comment by AndrewTriesToCode** [comment, marked as answer]
Hi, sorry I didn't respond sooner. The first registered strategy to find a matching tenant will "win" is in your case the route will match rather than the subdomain.

**Reply by vigouredelaruse** [reply]
thanks - and for future reference this implies that
- a hosting provider that allows host strategy tenant resolution should register the host strategy first, probably because the host namespace is a superset of http query path namespace
